### DISCUSSION TITLE
Figure justification in docx output

### DISCUSSION THREAD

**Discussion by mcneills** [quarto-dev/quarto-cli#801] [Q&A] [answered]
A couple of things I noticed in Quarto:

1. The Quarto documentation suggests that it is possible to left, centre (or center), or right justify figures with a .docx output (see [Quarto figures](https://quarto.org/docs/reference/formats/docx.html#figures)). The following Quarto fragment tries a right justification, which does not work. Neither does centre justification.

2. Specifying the figure height and/or width works more-or-less as advertised, except the result is scaled 101% in the docx. Changing the scaling to 100% in the docx file produces the correct result (or, as close as numerically reasonable). 

One or both of these might be Pandoc related, but I thought I would start here...

Stephen

```
---
title: "This is the title"
format: docx
page-width: 7.874016
editor: visual
author: Stephen McNeill
date: 5 May 2022
execute:
  echo: false
  warning: false
  error: false
---

## A Word figure

Quarto allows more control over figures. Here is a test:

```{r}
#| label: fig-example
#| fig-cap: "A rather trivial plot"
#| fig-align: right
#| fig-width: 3.50
#| fig-height: 5.0
x <- rnorm(100)
y <- rnorm(100)
plot(x,y)
```

**Comment by jjallaire** [comment, marked as answer]
Thanks for reporting this! An oversight. Should be fixed here: https://github.com/quarto-dev/quarto-cli/commit/212ad58f175cc3a2a3ee878737a2e416b1a7139e (and available in the next build which should be out in ~ 1hr)

**Comment by baptiste** [comment]
I'm having trouble with this when the figure has no caption (/label?):

````
---
title: "Untitled"
format: 
  docx: default
---

This is fine:

```{r}
#| label: fig-example
#| fig-align: right
#| fig-width: 3.50
#| echo: false
download.file('https://upload.wikimedia.org/wikipedia/en/thumb/8/80/Wikipedia-logo-v2.svg/103px-Wikipedia-logo-v2.svg.png','logo.png')
knitr::include_graphics("logo.png")
```

This is not:

```{r}
#| fig-align: right
#| fig-width: 3.50
#| echo: false
knitr::include_graphics("logo.png")
```

Same issue outside chunks:

![](logo.png){width=40% fig-align="right"}


````

<img width="755" alt="Screenshot 2024-03-27 at 20 39 31" src="https://github.com/quarto-dev/quarto-cli/assets/18970/22b3811f-2d36-4dab-814b-65c277540202">

> quarto -v
1.5.26

**Reply by cderv** [reply]
This is related to 
* https://github.com/quarto-dev/quarto-cli/issues/6509

The cross reference overhaul changed the behavior on this, where `fig-align` works for figures which are identified by label with `#fig-` prefix

Follow discussion there for now.

**Reply by baptiste** [reply]
Thanks – in the meantime this seems to work for my very limited use-case (just adding one centered image at the top of a word doc). I struggled a bit to find the `from: markdown+/-extension` trick, I think it'd be helpful to have the docs return this tip when searching for pandoc extensions like these.

````
---
title: "Untitled"
format: 
  docx: default
from: markdown-implicit_figures
---

![&nbsp;](logo.png){width=40% fig-align="center"}


````
